src/google_map.js

```diff
@@ -297,6 +297,10 @@ export default class GoogleMap extends Component {
   // otherwise there is no homeomorphism between screen coordinates and map
   // (one map coordinate can have different screen coordinates)
   _getMinZoom = () => {
+    if (this.options.minZoomOverride) {
```

Comment:
Thank you for your PR, but I think
this logic must be moved outside of this function, 
because with this logic it returns not a min possible zoom, but `minZoomOrSomethingFromOptions`. 

`this.options` must be removed, as it does not needed.